### PR TITLE
docs: update to include Supabase S3 File Provider

### PR DESCRIPTION
**Pull request by 420coupe**
update to include instructions for Supabase S3 Storage

### CODE REVIEW

File: www/apps/resources/app/architectural-modules/file/s3/page.mdx [hunk 1]
```diff
@@ -18,6 +18,7 @@ The S3 File Module Provider integrates Amazon S3 and services following a compat
     <TabsTrigger value="aws">AWS S3</TabsTrigger>
     <TabsTrigger value="minio">MinIO</TabsTrigger>
     <TabsTrigger value="spaces">DigitalOcean Spaces</TabsTrigger>
+    <TabsTrigger value="suapbase">Supabase S3 Storage</TabsTrigger>
```
Comment (shahednasser) [inline www/apps/resources/app/architectural-modules/file/s3/page.mdx]:
```suggestion
    <TabsTrigger value="supabase">Supabase S3 Storage</TabsTrigger>
```

File: www/apps/resources/app/architectural-modules/file/s3/page.mdx [hunk 2]
```diff
@@ -82,6 +83,24 @@ The S3 File Module Provider integrates Amazon S3 and services following a compat
     - Create [DigitalOcean Spaces bucket](https://docs.digitalocean.com/products/spaces/how-to/create/).
     - Create [DigitalOcean Spaces access and secret access keys](https://docs.digitalocean.com/products/spaces/how-to/manage-access/#access-keys).
 
+    </TabsContent>
+<TabsContent value="supabase">
```
Comment (shahednasser) [inline www/apps/resources/app/architectural-modules/file/s3/page.mdx]:
```suggestion
    <TabsContent value="supabase">
```

File: www/apps/resources/app/architectural-modules/file/s3/page.mdx [hunk 3]
```diff
@@ -82,6 +83,24 @@ The S3 File Module Provider integrates Amazon S3 and services following a compat
...
+
+    - [Supabase account](https://supabase.com/dashboard/sign-in?returnTo=%2Fprojects).
```
Comment (shahednasser) [inline www/apps/resources/app/architectural-modules/file/s3/page.mdx]:
```suggestion
    - [Supabase account](https://supabase.com/dashboard/sign-in) with a project.
```

File: www/apps/resources/app/architectural-modules/file/s3/page.mdx [hunk 4]
```diff
@@ -82,6 +83,24 @@ The S3 File Module Provider integrates Amazon S3 and services following a compat
...
+    - Create [New Project]
```
Comment (shahednasser) [inline www/apps/resources/app/architectural-modules/file/s3/page.mdx]:
```suggestion
```

File: www/apps/resources/app/architectural-modules/file/s3/page.mdx [hunk 5]
```diff
@@ -82,6 +83,24 @@ The S3 File Module Provider integrates Amazon S3 and services following a compat
...
+    - Create [New Public S3 bucket](https://supabase.com/docs/guides/storage/buckets/creating-buckets?queryGroups=language&language=dashboard).
+    - Create [New S3 Access Keys](https://supabase.com/docs/guides/storage/s3/authentication?queryGroups=language&language=javascript#s3-access-keys).
+    - Create [Storage Policy](https://supabase.com/docs/guides/storage/security/access-control).
+      1. On your bucket's dashboard, under configuration, click polocies.
```
Comment (shahednasser) [inline www/apps/resources/app/architectural-modules/file/s3/page.mdx]:
```suggestion
      1. On your bucket's dashboard, click on Policies under the Configuration sidebar.
```

File: www/apps/resources/app/architectural-modules/file/s3/page.mdx [hunk 6]
```diff
@@ -82,6 +83,24 @@ The S3 File Module Provider integrates Amazon S3 and services following a compat
...
+      2. Click on New Policy under Other policies under storage.objects.
```
Comment (shahednasser) [inline www/apps/resources/app/architectural-modules/file/s3/page.mdx]:
```suggestion
      2. Click on New Policy under "Other policies under storage.objects".
```

File: www/apps/resources/app/architectural-modules/file/s3/page.mdx [hunk 7]
```diff
@@ -82,6 +83,24 @@ The S3 File Module Provider integrates Amazon S3 and services following a compat
...
+      3. Click get started quickly and enable insert access for authenticated users only.
```
Comment (shahednasser) [inline www/apps/resources/app/architectural-modules/file/s3/page.mdx]:
```suggestion
      3. Click get started quickly, choose "insert access for authenticated users only", and click the "Use this template" button.
```

File: www/apps/resources/app/architectural-modules/file/s3/page.mdx [hunk 8]
```diff
@@ -82,6 +83,24 @@ The S3 File Module Provider integrates Amazon S3 and services following a compat
...
+      4. Click ALL instead of INSERT, then review and save policy.
```
Comment (shahednasser) [inline www/apps/resources/app/architectural-modules/file/s3/page.mdx]:
```suggestion
      4. Click ALL for "Allowed Operations", click the "Review" button, then the "Save policy" button.
```

File: www/apps/resources/app/architectural-modules/file/s3/page.mdx [hunk 9]
```diff
@@ -82,6 +83,24 @@ The S3 File Module Provider integrates Amazon S3 and services following a compat
...
+      5. Click settings under configuration.
+      6. Click New access key.
+      7. Name new access key, and save id and secret key. 
+      8. S3_FILE_URL= ``https://{uniqueID}/supabase.co/storage/v1/object/public``
+      9. Copy settings for ``S3_ENDPOINT``, ``S3_REGION``, ``S3_BUCKET``, ``S3_ACCESS_KEY``, ``S3_SECRET_KEY``
```
Comment (shahednasser) [inline www/apps/resources/app/architectural-modules/file/s3/page.mdx]:
```suggestion
```

No need for these as you already added the link to the S3 access key guide above.

File: www/apps/resources/app/architectural-modules/file/s3/page.mdx [hunk 10]
```diff
@@ -252,6 +272,7 @@ module.exports = {
       - For AWS S3, the endpoint is of the format `https://s3.{region}.amazonaws.com`
       - For MinIO, it's the URL to the MinIO server.
       - For DigitalOcean Spaces, it's the Spaces Origin Endpoint.
+      - For Supabase, it's the Endpoint url
```
Comment (shahednasser) [inline www/apps/resources/app/architectural-modules/file/s3/page.mdx]:
```suggestion
      - For Supabase, it's the Endpoint URL in the [Storage Settings](https://supabase.com/docs/guides/storage/s3/authentication?queryGroups=language&language=javascript#s3-access-keys).
```

File: www/apps/resources/app/architectural-modules/file/s3/page.mdx [hunk 11]
```diff
@@ -161,6 +180,7 @@ module.exports = {
       - For AWS S3, the endpoint is of the format `https://{bucket}.s3.{region}.amazonaws.com`
       - For MinIO, it's the URL to the MinIO server with the bucket's name. For example, `https://{server-domain}/{bucket}`.
       - For DigitalOcean Spaces, it's either the Origin Endpoint or the CDN endpoint of your Spaces Object Storage bucket.
+      - for Supabase, it's `https://{uniqueID}.supabase.co/storage/v1/object/public/{bucket}`.
```
Comment (shahednasser) [inline www/apps/resources/app/architectural-modules/file/s3/page.mdx]:
```suggestion
      - for Supabase, it's `https://{uniqueID}.supabase.co/storage/v1/object/public/{bucket}`. You can retrieve the `uniqueID` from [Storage Settings](https://supabase.com/docs/guides/storage/s3/authentication?queryGroups=language&language=javascript#s3-access-keys) page in the Endpoint URL.
```

**Review by shahednasser — CHANGES_REQUESTED** [review]
Hi @420coupe thank you for this contribution!

I tried following along the steps but it seems like there's a compatibility issue between the S3 file provider and supabase's storage. Did you get it working? If so, can you share what your configurations look like (without secrets, just want to make sure that the configurations I used were correct).

I also added comments for changes text-wise.

**Comment by 420coupe** [comment]
> Hi @420coupe thank you for this contribution!
> 
> I tried following along the steps but it seems like there's a compatibility issue between the S3 file provider and supabase's storage. Did you get it working? If so, can you share what your configurations look like (without secrets, just want to make sure that the configurations I used were correct).
> 
> I also added comments for changes text-wise.

Yes, was able to get this working. 

#medusa-config.js

```js
[Modules.FILE]: {
      resolve: "@medusajs/file",
      options: {
        providers: [
          {
            resolve: "@medusajs/file-s3",
            id: "s3",
            options: {
              file_url: process.env.SUPABASE_S3_FILE_URL,
              access_key_id: process.env.SUPABASE_S3_ACCESS_KEY,
              secret_access_key: process.env.SUPABASE_S3_SECRET_KEY,
              region: process.env.SUPABASE_S3_REGION,
              bucket: process.env.SUPABASE_S3_BUCKET,
              endpoint: process.env.SUPABASE_S3_ENDPOINT,
              additional_client_config: {
                forcePathStyle: true,
              },
            },
          },
        ],
      },
    },
```
    
#.env
  



  ```js
  SUPABASE_S3_FILE_URL=https://{uniqueIdFromEndpointUrl}.supabase.co/storage/v1/object/public/{bucketName}
SUPABASE_S3_ACCESS_KEY={accessKey}
SUPABASE_S3_SECRET_KEY={secretKey}
SUPABASE_S3_ENDPOINT={endpointUrl}
SUPABASE_S3_REGION={region}
SUPABASE_S3_BUCKET=[bucketName}
```

**Review by shahednasser — CHANGES_REQUESTED** [review]
Adding `forcePathStyle` fixed it, and someone else reported something similar for minio.

Can you add a section before troubleshooting called "Additional Configuration for MinIO and Supabase" mentioning that it's important to add this config for those two storage services (with a code snippet of the option)?

**Review by shahednasser — CHANGES_REQUESTED** [review]
Thanks for the change! But I meant to add the following before `## Troubleshooting`:

~~~

## Additional Configuration for MinIO and Supabase

If you're using MinIO or Supabase, set `forcePathStyle` to `true` in the `additional_client_config` object.

For example:

```js title="medusa-config.js"
module.exports = defineConfig({
  // ...
  modules: {
    [Modules.FILE]: {
      resolve: "@medusajs/file",
      options: {
        providers: [
          {
            resolve: "@medusajs/file-s3",
            id: "s3",
            options: {
              // ...
              additional_client_config: {
                forcePathStyle: true,
              },
            },
          },
        ],
      },
    },
  }
})
```

---

~~~

And then remove the new troubleshooting file you added.

**Comment by 420coupe** [comment]
> Thanks for the change! But I meant to add the following before `## Troubleshooting`:
> 
> ```
> 
> ## Additional Configuration for MinIO and Supabase
> 
> If you're using MinIO or Supabase, set `forcePathStyle` to `true` in the `additional_client_config` object.
> 
> For example:
> 
> ```js title="medusa-config.js"
> module.exports = defineConfig({
>   // ...
>   modules: {
>     [Modules.FILE]: {
>       resolve: "@medusajs/file",
>       options: {
>         providers: [
>           {
>             resolve: "@medusajs/file-s3",
>             id: "s3",
>             options: {
>               // ...
>               additional_client_config: {
>                 forcePathStyle: true,
>               },
>             },
>           },
>         ],
>       },
>     },
>   }
> })
> ```
> 
> ---
> ```
> 
> And then remove the new troubleshooting file you added.

think i got it now.. maybe

File: www/apps/resources/app/architectural-modules/file/s3/page.mdx [hunk 12]
```diff
@@ -339,6 +349,37 @@ module.exports = {
...
+                forcePathStyle: true,
+              },
+            },
+          },
+        ],
+      },
+    },
+  }
+})
+```
```
Comment (shahednasser) [inline www/apps/resources/app/architectural-modules/file/s3/page.mdx:381]:
~~~suggestion
```
---

~~~

**Review by shahednasser — APPROVED** [review]
One last change and we're ready to go 😄 I'll commit the suggestion

**Comment by shahednasser** [comment]
Thank you for your contribution @420coupe !

**Pull request merged**